Binary files poco-2009.1-tudresden/.DS_Store and poco-2009.1-tudresden-cmake/.DS_Store differ
diff -rupN poco-2009.1-tudresden/CMakeLists.txt poco-2009.1-tudresden-cmake/CMakeLists.txt
--- poco-2009.1-tudresden/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/CMakeLists.txt	2013-04-11 11:48:14.000000000 +0200
@@ -0,0 +1,100 @@
+project(POCO)
+cmake_minimum_required(VERSION 2.8)
+
+option(USE_CXX11 "Build with explicit c++ 11 support under GCC/LLVM" ON)
+option(POCO_NO_TESTS "Do not build testsuites." OFF)
+option(POCO_NO_SAMPLES "Do not build samples." OFF)
+option(POCO_NO_WSTRING "Compile with -DPOCO_NO_WSTRING. Useful if your C++ compiler does not support std::wstring (such as uClibc-based systems)." OFF)
+option(POCO_NO_FPENVIRONMENT "Compile with -DPOCO_NO_FPENVIRONMENT. Useful if your C++ compiler has incomplete floating-point support (such as uClibc-based systems).")
+option(POCO_POQUITO "Omit a few features for smaller codesize when linking statically for embedded targets.")
+
+option(POCO_ENABLE_MACOSX_FRAMEWORKS "Causes POCO Libraries to be built as OS X Frameworks" OFF)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARCHFLAGS}")
+
+if(POCO_ENABLE_MACOSX_FRAMEWORKS)
+	set(HEADER_INSTALL_DESTINATION Headers)
+	set(CMAKE_INSTALL_NAME_DIR @executable_path/../Frameworks)
+else()
+	set(HEADER_INSTALL_DESTINATION include)
+#	set(CMAKE_INSTALL_NAME_DIR @rpath)
+endif()
+if(${USE_CXX11})
+	if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" 
+	OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang"
+	OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
+	message(STATUS "Checking for C++11 / C++0x availability")
+	include(CheckCXXCompilerFlag)
+	CHECK_CXX_COMPILER_FLAG(-std=c++11 HAS_CXX11)
+	if(HAS_CXX11)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+		if(${CMAKE_SYSTEM_NAME} STREQUAL Darwin)
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+			set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD "c++11")
+			set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY "libc++")
+		endif()
+		message(STATUS "C++11 found.")
+	else(HAS_CXX11)
+		message(STATUS "C++11 not found, checking C++0x.")
+		CHECK_CXX_COMPILER_FLAG(-std=c++0x HAS_CXX0X)
+		if(HAS_CXX0X)
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+			if(${CMAKE_SYSTEM_NAME} STREQUAL Darwin)
+				set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+				set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD "c++0x")
+				set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY "libc++")
+			endif()
+			message(STATUS "C++0x found.")
+		else()
+			message(FATAL_ERROR "Neither C++11 nor C++0x found.")
+		endif()
+	endif()
+endif()
+endif()
+if(${POCO_NO_WSTRING})
+	add_definitions(-DPOCO_NO_WSTRING)
+endif()
+if(${POCO_NO_FPENVIRONMENT})
+	add_definitions(-DPOCO_NO_FPENVIRONMENT)
+endif()
+if(${POCO_POQUITO})
+	add_definitions(-DPOCO_NO_FILECHANNEL -DPOCO_NO_SPLITTERCHANNEL -DPOCO_NO_SYSLOGCHANNEL -DDPOCO_UTIL_NO_INIFILECONFIGURATION -DDPOCO_UTIL_NO_XMLCONFIGURATION)
+endif()
+
+set(CMAKE_DEBUG_POSTFIX "d")
+set_directory_properties(PROPERTIES COMPILE_DEFINITIONS_DEBUG "_DEBUG=${DEBUGLEVEL}")
+set_directory_properties(PROPERTIES COMPILE_DEFINITIONS_RELEASE "NDEBUG")
+if(CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
+	set_directory_properties(PROPERTIES COMPILE_DEFINITIONS "_DEBUG=${DEBUGLEVEL}")
+else()
+	set_directory_properties(PROPERTIES COMPILE_DEFINITIONS "NDEBUG")
+endif()
+
+file(STRINGS libversion LIBVERSION)
+set(POCO_OSP_VERSION 2)
+add_definitions(-DPOCO_NO_TEMPLATE_ICOMPARE) #inhibits build error in NetSSL_OpenSSL referencing String.h in Foundations
+
+file(STRINGS components POCO_COMPONENTS)
+
+# TODO reduce this list by implementing the rest of the poco build in cmake
+list(APPEND OMIT 
+	Data/ODBC
+	Data/SQLite
+	Data/MySQL
+	OSP/BundleAdmin
+	OSP/Core
+	OSP/Data
+	OSP/Data/ODBC
+	OSP/Data/SQLite
+	OSP/Data/MySQL
+	OSP/Net
+	OSP/NetSSL_OpenSSL
+	OSP/SecureWebServer
+	OSP/WebServer
+	OSP/Shell
+	OSP/Shell/Client
+)
+list(REMOVE_ITEM POCO_COMPONENTS ${OMIT})
+foreach(POCO_COMPONENT ${POCO_COMPONENTS})
+	add_subdirectory(${POCO_COMPONENT})
+endforeach()
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/CppUnit/CMakeLists.txt poco-2009.1-tudresden-cmake/CppUnit/CMakeLists.txt
--- poco-2009.1-tudresden/CppUnit/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/CppUnit/CMakeLists.txt	2013-02-23 18:15:05.000000000 +0100
@@ -0,0 +1,23 @@
+include_directories(include src)
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(CppUnit SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(CppUnit
+	PROPERTIES
+	#FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION 1
+)
+
+install(TARGETS CppUnit 
+	FRAMEWORK DESTINATION frameworks
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION lib
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
+
+if(WIN32)
+	add_subdirectory(WinTestRunner)
+endif()
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/CppUnit/WinTestRunner/CMakeLists.txt poco-2009.1-tudresden-cmake/CppUnit/WinTestRunner/CMakeLists.txt
--- poco-2009.1-tudresden/CppUnit/WinTestRunner/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/CppUnit/WinTestRunner/CMakeLists.txt	2013-02-23 18:08:05.000000000 +0100
@@ -0,0 +1,19 @@
+include_directories(include src
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(WinTestRunner SHARED ${SOURCES} ${INCLUDES} )
+
+set_target_properties(WinTestRunner
+	PROPERTIES
+	#FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION 1
+)
+
+install(TARGETS WinTestRunner 
+	FRAMEWORK DESTINATION frameworks
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION lib
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/Crypto/CMakeLists.txt poco-2009.1-tudresden-cmake/Crypto/CMakeLists.txt
--- poco-2009.1-tudresden/Crypto/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/Crypto/CMakeLists.txt	2013-02-23 17:13:45.000000000 +0100
@@ -0,0 +1,24 @@
+include_directories(include src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoCrypto SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoCrypto PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${LIBVERSION}
+)
+
+find_package(OpenSSL REQUIRED)
+
+target_link_libraries(PocoCrypto PocoFoundation ${OPENSSL_LIBRARIES})
+
+install(TARGETS PocoCrypto
+	FRAMEWORK DESTINATION frameworks
+	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/Data/CMakeLists.txt poco-2009.1-tudresden-cmake/Data/CMakeLists.txt
--- poco-2009.1-tudresden/Data/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/Data/CMakeLists.txt	2013-02-23 18:11:07.000000000 +0100
@@ -0,0 +1,22 @@
+include_directories(include src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoData SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoData PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${LIBVERSION}
+)
+
+target_link_libraries(PocoData PocoFoundation)
+
+install(TARGETS PocoData
+	FRAMEWORK DESTINATION frameworks
+	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/Foundation/CMakeLists.txt poco-2009.1-tudresden-cmake/Foundation/CMakeLists.txt
--- poco-2009.1-tudresden/Foundation/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/Foundation/CMakeLists.txt	2013-02-23 16:24:01.000000000 +0100
@@ -0,0 +1,84 @@
+include_directories(include)
+include_directories(src)
+
+file(GLOB_RECURSE INCLUDES include/*.h)
+
+aux_source_directory(src SOURCES)
+
+if(NOT MINGW)
+	list(REMOVE_ITEM SOURCES
+		src/WindowsConsoleChannel.cpp
+		src/EventLogChannel.cpp
+	)
+else()
+	list(REMOVE_ITEM SOURCES
+		src/SysLogChannel.cpp
+	)
+endif()
+if(NOT VMS)
+	list(REMOVE_ITEM SOURCES
+		src/OpcomChannel.cpp
+	)
+endif()
+
+set(ZLIB_SOURCES
+	src/adler32.c
+	src/compress.c
+	src/crc32.c
+	src/deflate.c
+	src/gzio.c
+	src/infback.c
+	src/inffast.c
+	src/inflate.c
+	src/inftrees.c
+	src/trees.c
+	src/zutil.c
+)
+
+set(PCRE_SOURCES
+	src/pcre_chartables.c
+	src/pcre_compile.c
+	src/pcre_exec.c
+	src/pcre_fullinfo.c
+	src/pcre_globals.c
+	src/pcre_maketables.c
+	src/pcre_newline.c
+	src/pcre_ord2utf8.c
+	src/pcre_study.c
+	src/pcre_tables.c
+	src/pcre_try_flipped.c
+	src/pcre_ucd.c
+	src/pcre_valid_utf8.c
+	src/pcre_xclass.c
+)
+
+
+list(REMOVE_ITEM SOURCES
+	${PCRE_SOURCES} ${ZLIB_SOURCES}
+)
+set(platform_regex "(.*)_(UNIX|VMS|WIN32|WIN32U|WINCE|POSIX|C99|DEC|DUMMY|SUN|VX|ANDROID|STD|HPUX)(.*)" )
+foreach(SOURCE_FILE ${SOURCES})
+	if(${SOURCE_FILE} MATCHES ${platform_regex})
+		#message(${SOURCE_FILE})
+		set_source_files_properties(${SOURCE_FILE} PROPERTIES HEADER_FILE_ONLY true)
+	endif()
+endforeach()
+
+add_library(zlib STATIC ${ZLIB_SOURCES})
+add_library(PCRE STATIC ${PCRE_SOURCES})
+add_library(PocoFoundation SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoFoundation
+	PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${LIBVERSION}
+)
+
+target_link_libraries(PocoFoundation zlib PCRE)
+
+install(TARGETS PocoFoundation
+	FRAMEWORK DESTINATION frameworks
+	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+)
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/Foundation/src/NumberFormatter.cpp poco-2009.1-tudresden-cmake/Foundation/src/NumberFormatter.cpp
--- poco-2009.1-tudresden/Foundation/src/NumberFormatter.cpp	2009-05-25 16:44:27.000000000 +0200
+++ poco-2009.1-tudresden-cmake/Foundation/src/NumberFormatter.cpp	2013-07-07 13:42:46.000000000 +0200
@@ -241,7 +241,7 @@ void NumberFormatter::appendHex(std::str
 void NumberFormatter::append(std::string& str, Int64 value)
 {
 	char buffer[64];
-	std::sprintf(buffer, "%"I64_FMT"d", value);
+	std::sprintf(buffer, "%" I64_FMT"d", value);
 	str.append(buffer);
 }
 
@@ -251,7 +251,7 @@ void NumberFormatter::append(std::string
 	poco_assert (width > 0 && width < 64);
 
 	char buffer[64];
-	std::sprintf(buffer, "%*"I64_FMT"d", width, value);
+	std::sprintf(buffer, "%*" I64_FMT"d", width, value);
 	str.append(buffer);
 }
 
@@ -261,7 +261,7 @@ void NumberFormatter::append0(std::strin
 	poco_assert (width > 0 && width < 64);
 
 	char buffer[64];
-	std::sprintf(buffer, "%0*"I64_FMT"d", width, value);
+	std::sprintf(buffer, "%0*" I64_FMT"d", width, value);
 	str.append(buffer);
 }
 
@@ -269,7 +269,7 @@ void NumberFormatter::append0(std::strin
 void NumberFormatter::appendHex(std::string& str, Int64 value)
 {
 	char buffer[64];
-	std::sprintf(buffer, "%"I64_FMT"X", value);
+	std::sprintf(buffer, "%" I64_FMT"X", value);
 	str.append(buffer);
 }
 
@@ -279,7 +279,7 @@ void NumberFormatter::appendHex(std::str
 	poco_assert (width > 0 && width < 64);
 
 	char buffer[64];
-	std::sprintf(buffer, "%0*"I64_FMT"X", width, value);
+	std::sprintf(buffer, "%0*" I64_FMT"X", width, value);
 	str.append(buffer);
 }
 
@@ -287,7 +287,7 @@ void NumberFormatter::appendHex(std::str
 void NumberFormatter::append(std::string& str, UInt64 value)
 {
 	char buffer[64];
-	std::sprintf(buffer, "%"I64_FMT"u", value);
+	std::sprintf(buffer, "%" I64_FMT"u", value);
 	str.append(buffer);
 }
 
@@ -297,7 +297,7 @@ void NumberFormatter::append(std::string
 	poco_assert (width > 0 && width < 64);
 
 	char buffer[64];
-	std::sprintf(buffer, "%*"I64_FMT"u", width, value);
+	std::sprintf(buffer, "%*" I64_FMT"u", width, value);
 	str.append(buffer);
 }
 
@@ -307,7 +307,7 @@ void NumberFormatter::append0(std::strin
 	poco_assert (width > 0 && width < 64);
 
 	char buffer[64];
-	std::sprintf(buffer, "%0*"I64_FMT"u", width, value);
+	std::sprintf(buffer, "%0*" I64_FMT"u", width, value);
 	str.append(buffer);
 }
 
@@ -315,7 +315,7 @@ void NumberFormatter::append0(std::strin
 void NumberFormatter::appendHex(std::string& str, UInt64 value)
 {
 	char buffer[64];
-	std::sprintf(buffer, "%"I64_FMT"X", value);
+	std::sprintf(buffer, "%" I64_FMT"X", value);
 	str.append(buffer);
 }
 
@@ -325,7 +325,7 @@ void NumberFormatter::appendHex(std::str
 	poco_assert (width > 0 && width < 64);
 
 	char buffer[64];
-	std::sprintf(buffer, "%0*"I64_FMT"X", width, value);
+	std::sprintf(buffer, "%0*" I64_FMT"X", width, value);
 	str.append(buffer);
 }
 
@@ -373,7 +373,7 @@ void NumberFormatter::append(std::string
 {
 	char buffer[24];
 #if defined(POCO_PTR_IS_64_BIT)
-	std::sprintf(buffer, "%016"I64_FMT"X", (UIntPtr) ptr);
+	std::sprintf(buffer, "%016" I64_FMT"X", (UIntPtr) ptr);
 #else
 	std::sprintf(buffer, "%08lX", (UIntPtr) ptr);
 #endif
diff -rupN poco-2009.1-tudresden/Foundation/src/NumberParser.cpp poco-2009.1-tudresden-cmake/Foundation/src/NumberParser.cpp
--- poco-2009.1-tudresden/Foundation/src/NumberParser.cpp	2009-05-25 16:44:27.000000000 +0200
+++ poco-2009.1-tudresden-cmake/Foundation/src/NumberParser.cpp	2013-04-11 11:53:22.000000000 +0200
@@ -119,7 +119,7 @@ Int64 NumberParser::parse64(const std::s
 bool NumberParser::tryParse64(const std::string& s, Int64& value)
 {
 	char temp;
-	return std::sscanf(s.c_str(), "%"I64_FMT"d%c", &value, &temp) == 1;
+	return std::sscanf(s.c_str(), "%" I64_FMT"d%c", &value, &temp) == 1;
 }
 
 
@@ -136,7 +136,7 @@ UInt64 NumberParser::parseUnsigned64(con
 bool NumberParser::tryParseUnsigned64(const std::string& s, UInt64& value)
 {
 	char temp;
-	return std::sscanf(s.c_str(), "%"I64_FMT"u%c", &value, &temp) == 1;
+	return std::sscanf(s.c_str(), "%" I64_FMT"u%c", &value, &temp) == 1;
 }
 
 
@@ -153,7 +153,7 @@ UInt64 NumberParser::parseHex64(const st
 bool NumberParser::tryParseHex64(const std::string& s, UInt64& value)
 {
 	char temp;
-	return std::sscanf(s.c_str(), "%"I64_FMT"x%c", &value, &temp) == 1;
+	return std::sscanf(s.c_str(), "%" I64_FMT"x%c", &value, &temp) == 1;
 }
 
 
diff -rupN poco-2009.1-tudresden/Net/CMakeLists.txt poco-2009.1-tudresden-cmake/Net/CMakeLists.txt
--- poco-2009.1-tudresden/Net/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/Net/CMakeLists.txt	2013-02-23 17:05:45.000000000 +0100
@@ -0,0 +1,23 @@
+include_directories(include src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoNet SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoNet PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	DEFINE_SYMBOL "Net_EXPORTS"
+	VERSION ${LIBVERSION}
+)
+
+target_link_libraries(PocoNet PocoFoundation)
+
+install(TARGETS PocoNet
+	FRAMEWORK DESTINATION frameworks
+	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/NetSSL_OpenSSL/CMakeLists.txt poco-2009.1-tudresden-cmake/NetSSL_OpenSSL/CMakeLists.txt
--- poco-2009.1-tudresden/NetSSL_OpenSSL/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/NetSSL_OpenSSL/CMakeLists.txt	2013-02-23 17:43:38.000000000 +0100
@@ -0,0 +1,27 @@
+include_directories(include src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+include_directories(${POCO_SOURCE_DIR}/Crypto/include)
+include_directories(${POCO_SOURCE_DIR}/Util/include)
+include_directories(${POCO_SOURCE_DIR}/Net/include)
+
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoNetSSL SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoNetSSL PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${LIBVERSION}
+)
+
+find_package(OpenSSL REQUIRED)
+
+target_link_libraries(PocoNetSSL PocoFoundation PocoCrypto PocoUtil PocoNet ${OPENSSL_LIBRARIES})
+
+install(TARGETS PocoNetSSL
+	FRAMEWORK DESTINATION frameworks
+	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/OSP/BundleCreator/CMakeLists.txt poco-2009.1-tudresden-cmake/OSP/BundleCreator/CMakeLists.txt
--- poco-2009.1-tudresden/OSP/BundleCreator/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/OSP/BundleCreator/CMakeLists.txt	2013-03-04 16:31:36.000000000 +0100
@@ -0,0 +1,16 @@
+include_directories(src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+include_directories(${POCO_SOURCE_DIR}/Util/include)
+include_directories(${POCO_SOURCE_DIR}/Zip/include)
+include_directories(${POCO_SOURCE_DIR}/XML/include)
+include_directories(${POCO_SOURCE_DIR}/OSP/include)
+
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES src/*.h) 
+
+add_executable(bundle ${SOURCES} ${INCLUDES})
+target_link_libraries(bundle PocoOSP PocoFoundation PocoUtil PocoZip PocoXML)
+
+install(TARGETS bundle
+	RUNTIME DESTINATION bin
+)
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/OSP/BundleSign/CMakeLists.txt poco-2009.1-tudresden-cmake/OSP/BundleSign/CMakeLists.txt
--- poco-2009.1-tudresden/OSP/BundleSign/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/OSP/BundleSign/CMakeLists.txt	2013-07-07 13:11:29.000000000 +0200
@@ -0,0 +1,29 @@
+include_directories(include)
+include_directories(src)
+include_directories(${POCO_SOURCE_DIR}/OSP/include)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+include_directories(${POCO_SOURCE_DIR}/Util/include)
+include_directories(${POCO_SOURCE_DIR}/Zip/include)
+include_directories(${POCO_SOURCE_DIR}/XML/include)
+include_directories(${POCO_SOURCE_DIR}/Crypto/include)
+
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoOSPBundleSign SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoOSPBundleSign
+	PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${POCO_OSP_VERSION}
+)
+
+target_link_libraries(PocoOSPBundleSign PocoOSP PocoZip PocoCrypto PocoUtil PocoXML PocoFoundation)
+
+install(TARGETS PocoOSPBundleSign 
+	FRAMEWORK DESTINATION frameworks
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION lib
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/OSP/CMakeLists.txt poco-2009.1-tudresden-cmake/OSP/CMakeLists.txt
--- poco-2009.1-tudresden/OSP/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/OSP/CMakeLists.txt	2013-02-23 16:26:05.000000000 +0100
@@ -0,0 +1,27 @@
+include_directories(include)
+include_directories(src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+include_directories(${POCO_SOURCE_DIR}/Util/include)
+include_directories(${POCO_SOURCE_DIR}/Zip/include)
+include_directories(${POCO_SOURCE_DIR}/XML/include)
+
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoOSP SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoOSP
+	PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${POCO_OSP_VERSION}
+)
+
+target_link_libraries(PocoOSP PocoFoundation PocoUtil PocoZip PocoXML)
+
+install(TARGETS PocoOSP 
+	FRAMEWORK DESTINATION frameworks
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION lib
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/OSP/Web/CMakeLists.txt poco-2009.1-tudresden-cmake/OSP/Web/CMakeLists.txt
--- poco-2009.1-tudresden/OSP/Web/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/OSP/Web/CMakeLists.txt	2013-07-07 13:10:38.000000000 +0200
@@ -0,0 +1,45 @@
+include_directories(include)
+include_directories(src)
+include_directories(${POCO_SOURCE_DIR}/OSP/include)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+include_directories(${POCO_SOURCE_DIR}/Util/include)
+include_directories(${POCO_SOURCE_DIR}/Zip/include)
+include_directories(${POCO_SOURCE_DIR}/XML/include)
+include_directories(${POCO_SOURCE_DIR}/Net/include)
+
+set(LIB_SOURCES
+	src/MediaTypeMapper.cpp
+	src/WebServerDispatcher.cpp 
+	src/WebServerRequestHandlerFactory.cpp 
+	src/WebServerExtensionPoint.cpp 
+	src/WebSession.cpp 
+	src/WebRequestHandlerFactory.cpp
+    src/WebServerRequestHandler.cpp 
+    src/WebSessionManager.cpp
+)
+
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoOSPWeb SHARED ${LIB_SOURCES} ${INCLUDES})
+
+set_target_properties(PocoOSPWeb
+	PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${POCO_OSP_VERSION}
+)
+
+target_link_libraries(PocoOSPWeb PocoFoundation PocoUtil PocoNet PocoXML PocoOSP)
+
+install(TARGETS PocoOSPWeb 
+	FRAMEWORK DESTINATION frameworks
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION lib
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
+
+add_library(osp.web SHARED src/WebBundleActivator.cpp)
+target_link_libraries(osp.web PocoOSPWeb PocoOSP PocoUtil PocoXML PocoFoundation)
+add_custom_command(TARGET osp.web POST_BUILD
+	bundle -o${CMAKE_CURRENT_BINARY_DIRECTORY}/../bundles ${CMAKE_CURRENT_SOURCE_DIRECTORY}/WebBundle.bndlspec
+)
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/Util/CMakeLists.txt poco-2009.1-tudresden-cmake/Util/CMakeLists.txt
--- poco-2009.1-tudresden/Util/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/Util/CMakeLists.txt	2013-02-23 16:24:51.000000000 +0100
@@ -0,0 +1,34 @@
+include_directories(include)
+include_directories(src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+include_directories(${POCO_SOURCE_DIR}/XML/include)
+
+aux_source_directory(src SOURCES)
+
+if(NOT WIN32)
+	list(REMOVE_ITEM SOURCES
+		src/WinRegistryConfiguration.cpp
+		src/WinRegistryKey.cpp
+		src/WinService.cpp
+	)
+endif()
+
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoUtil SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoUtil
+	PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${LIBVERSION}
+)
+
+target_link_libraries(PocoUtil PocoFoundation PocoXML)
+
+install(TARGETS PocoUtil
+	FRAMEWORK DESTINATION frameworks
+	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
diff -rupN poco-2009.1-tudresden/XML/CMakeLists.txt poco-2009.1-tudresden-cmake/XML/CMakeLists.txt
--- poco-2009.1-tudresden/XML/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/XML/CMakeLists.txt	2013-02-23 16:32:21.000000000 +0100
@@ -0,0 +1,24 @@
+include_directories(include)
+include_directories(src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+include_directories(${POCO_SOURCE_DIR}/Util/include)
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES "include" "include/*.h") 
+
+add_definitions(-DXML_NS -DXML_DTD -DHAVE_EXPAT_CONFIG_H)
+add_library(PocoXML SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoXML
+	PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${LIBVERSION}
+)
+
+target_link_libraries(PocoXML PocoFoundation)
+
+install(TARGETS PocoXML
+	FRAMEWORK DESTINATION frameworks
+	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+)
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
diff -rupN poco-2009.1-tudresden/Zip/CMakeLists.txt poco-2009.1-tudresden-cmake/Zip/CMakeLists.txt
--- poco-2009.1-tudresden/Zip/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ poco-2009.1-tudresden-cmake/Zip/CMakeLists.txt	2013-02-23 16:58:51.000000000 +0100
@@ -0,0 +1,22 @@
+include_directories(include src)
+include_directories(${POCO_SOURCE_DIR}/Foundation/include)
+
+aux_source_directory(src SOURCES)
+file(GLOB_RECURSE INCLUDES include/*.h) 
+
+add_library(PocoZip SHARED ${SOURCES} ${INCLUDES})
+
+set_target_properties(PocoZip PROPERTIES
+	FRAMEWORK ${POCO_ENABLE_MACOSX_FRAMEWORKS}
+	VERSION ${LIBVERSION}
+)
+
+target_link_libraries(PocoZip PocoFoundation)
+
+install(TARGETS PocoZip
+	FRAMEWORK DESTINATION frameworks
+	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+)
+
+install(DIRECTORY include/ DESTINATION ${HEADER_INSTALL_DESTINATION} FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
